build.assets/build-package.sh

```diff
@@ -68,9 +68,9 @@ VENDOR="Gravitational"
 DESCRIPTION="Teleport is a gateway for managing access to clusters of Linux servers via SSH or the Kubernetes API"
 DOCS_URL="https://goteleport.com/docs"
 
-# signing IDs to use for mac (must be pre-loaded into the keychain on the build box)
-DEVELOPER_ID_APPLICATION="Developer ID Application: Gravitational Inc." # used for signing binaries
-DEVELOPER_ID_INSTALLER="Developer ID Installer: Gravitational Inc." # used for signing packages
+# SHA1 fingerprints of certificates used for signing mac artifacts (certificates must be pre-loaded into the keychain on the build box)
```

Comment:
Is there a corresponding change on the buildbox that will make the change from identifying a cert by name to a identifying it via fingerprint work? Or does the signing tool just silently take care of this?

Reply:
Good question. Here's the relelevant text on signing identities from `man codesign`


```
SIGNING IDENTITIES

     To be used for code signing, a digital identity must be stored in a key-
     chain that is on the calling user's keychain search list.  All keychain
     sources are supported if properly configured. In particular, it is possi-
     ble to sign code with an identity stored on a supported smart card.  If
     your signing identity is stored in a different form, you need to make it
     available in keychain form to sign code with it.
     If the --keychain argument is used, identity is only looked-for in the
     specific keychain given. This is meant to help disambiguate references to
     identities.  Even in that case, the full keychain search list is still
     consulted for additional certificates needed to complete the signature.

     The identity is first considered as the full name of a keychain identity
     preference.  If such a preference exists, it directly names the identity
     used.  Otherwise, the identity is located by searching all keychains for
     a certificate whose subject common name (only) contains the identity
     string given. If there are multiple matches, the operation fails and no
     signing is performed; however, an exact match is preferred over a partial
     match.  These comparisons are case sensitive.  Multiple instances of the
     exactly same certificate in multiple keychains are tolerated as harmless.

     If identity consists of exactly forty hexadecimal digits, it is instead
     interpreted as the SHA-1 hash of the certificate part of the desired
     identity.  In this case, the identity's subject name is not considered.

     Both identity preferences and certificate hashes can be used to identify
     a particular signing identity regardless of name. Identity preferences
     are global settings for each user and provide a layer of indirection.
     Certificate hashes are very explicit and local. These choices, combined
     with what is placed into Xcode project and target build variables and/or
     script settings, allows for very flexible designation of signing identi-
     ties.
```

I chose to swap to the cert SHA1 fingerprint because it allows us to be crisp about which certificate is used for signing.  Using a string name results in much less specific behavior (as described above), especially when multiple certificates for the same signing identity are present in the keychain -- which is our current situation during the cutover so that both old and new codepaths work.

Long term, I'd like to approach this issue via https://github.com/gravitational/teleport/issues/8210, which should remove hardcoded refs to externally managed signing identities from the source code entirely -- we'd have a single floating ref (and environment variable).